recipe/meta.yaml

```
@@ -1,50 +1,94 @@
+    # and our CUDAtoolkit v11.8 doesn't have the compiler tools.
+    # Use these patches for PyTorch v2.4.0 and above.
+    #- patches/0001-do-not-package-third_party-folder.patch
+    #- patches/0005-Unvendor-third-party-libs.patch
+    #- patches/0008-Search-for-libs-in-CONDA_PREFIX-instead-of-third_par.patch
+    # These patches are for unvendoring LLVM.
+    # The version of LLVM used for triton v2.3.0 is an unpinned commit on LLVM project's
+    # main branch, so we can't use a conda package for this.
+    # These patches can be used to help unvendoring if this changes.
+    # - patches/0004-Avoid-using-outdated-FindLLVM.patch
+    # - patches/0007-Fix-TableGen-issues.patch
+    # This patch applies even when llvm is vendored-in.
+    - patches/0009-unpack-llvm-within-env.patch
 
 build:
-  number: 4
-  # TODO: windows support should be available from next version;
-  #       CPU-only support still under development
-  skip: true  # [win or cuda_compiler_version == "None"]
-  string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
+  number: 0
+  # Triton only currently supports linux, and is a GPU optimization tool.
+  # We only have a linux-64 GPU builder at the moment.
+  # It's primarily for PyTorch, and they only use it for linux-64/GPU.
+  skip: true  # [not (linux and x86_64)]
+  # the torch.compile feature in PyTorch isn't supported on python 3.12:
+  # https://github.com/pytorch/pytorch/blob/97ff6cfd9c86c5c09d7ce775ab64ec5c99230f5d/test/test_transformers.py#L3418
+  skip: true  # [py>=312]
+  # Put the cuda version variable back into the build string when we unvendor CUDA.
+  # For triton v2.3.0, CUDA v12.3 is vendored-in.
+  #string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
+  string: cuda123py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
 
 requirements:
   build:
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}
     - make
     - cmake
-    - mlir
-    - llvmdev
+    - ninja
+    {% if torch_commit_pin %}
+    - git
+    {% endif %}
   host:
     - python
     - pybind11
     - pip
-    - llvm
-    - libmlir
-    - zlib
   run:
     - python
     - filelock
-    - pytorch =*=cuda*
-    - lit
+    - numpy
+    # Triton compiles cuda kernels so needs the compiler toolchain at runtime.
+    #
+    # {{ compiler('cuda') }} provides libdevice, ptxas, cuda.h, and the include path to cuda.h.
+    # cuda-cuobjdump provides cuobjdump and nvdisasm.
+    # Currently we keep CUDAtookit vendored-in (see patches section for reasoning), but this should be un-commented
+    # when this changes.
+    #
+    #- {{ compiler('cuda') }}
+    #- cuda-cuobjdump
+    #
+    #  gcc is required whether the cuda tools are vendored-in or not, to support CUDA compilation. (nvcc is called via
+    #  gcc when it processes .cu files, which are c++ syntax extended by CUDA syntax).
+    - {{ compiler('cxx') }}
+    - zlib
```

```suggestion
    - zlib
  run_constrained:
    - cmake >=3.20
```
https://github.com/triton-lang/triton/blob/3f8d91bb17f6e7bc33dc995ae0860db89d351c7b/python/setup.py#L388C14-L388C25

---

Umm, I don't think that's needed. I don't think cmake is used at runtime.

---

Unless their cmake config files are only compatible with cmake 2.23+...

---

I don't think this is needed either, it looks like it supports lit during testing: https://github.com/triton-lang/triton/commit/7c7b769e379132492d2c751e9dd68c4212bc433e
I'm pretty sure it doesn't need it to build stuff during runtime